go/vt/vtgate/vindexes/cfc.go

```diff
@@ -244,7 +241,8 @@ func (vind *CFC) Map(ctx context.Context, vcursor VCursor, ids []sqltypes.Value)
 
 // PrefixVindex switches the vindex to prefix mode
 func (vind *CFC) PrefixVindex() SingleColumn {
-	return vind.prefixVindex
+	// prefixCFC is only used in 'LIKE' compare expressions.
+	return &prefixCFC{CFC: vind}
 }
```

Comment:
Everytime we call this `PrefixVindex()` function, it would return a new struct. Is that what we want?

Reply:
that is not ideal, let me re-write this part.